scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/remotes/postgres/GetSpecificationGoalsAction.java

```diff
@@ -48,7 +50,8 @@ public List<PostgresGoalRecord> get(final long specificationId) throws SQLExcept
       final var revision = resultSet.getLong("revision");
       final var name = resultSet.getString("name");
       final var definition = resultSet.getString("definition");
-      goals.add(new PostgresGoalRecord(id, revision, name, definition));
+      final var enabled = resultSet.getBoolean("enabled");
```

Comment:
🔶 I'd like to chat through the pros and cons of filtering on the `enabled` field in the `SynchronousSchedulerAgent` vs at other stages in the pipeline.

The options, as I see them are:
1. Filter in the `SynchronousSchedulerAgent` (as you have in this PR)
2. Filter in the `PostgresSpecificationRepository` (by adding a `filter` line before the `map` line [here](https://github.com/NASA-AMMOS/aerie/blob/8cdb2e757f489502647f54ee3510594360f57155/scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/remotes/postgres/PostgresSpecificationRepository.java#L52-L59))
3. Filter in the `GetSpecificationGoalsAction`, by updating the SQL query to only return enabled goals

### Option 1: Filter in `SynchronousSchedulerAgent`
Pros: 
- The data in the database is faithfully represented all the way through the workflow
- The "business logic" i.e. interpretation of the `enabled` field happens in the scheduler itself

Cons:
- Type errors or runtime errors in the scheduling-dsl-compiler will fail the scheduling run, regardless of whether the errors originated in an enabled or a disabled goal (this could also be addressed by separating the compilation workflow from the scheduling run workflow, but currently the two are coupled together)
  - (I put this as a Con because I think it's undesirable, but I acknowledge that this can be up for debate)
- I question, should `enabled` really be the concern of the scheduler? Is there anything the scheduler would ever do with a disabled goal, besides ignore it?

### Option 2: Filter in `PostgresSpecificationRepository`
Pros:
- The `PostgresGoalRecord` is still a faithful representation of the data in the database
- Disabled Goals will not be compiled, and thus cannot cause the scheduling run to fail

Cons:
- The enabled field is "hidden" from the scheduler (but I think this might actually be a "Pro")

### Option 3: Filter in SQL
Pros:
- Only the `GetSpecificationGoalsAction` and the table itself have to change - the rest of the code can be oblivious to the `enabled` field

Cons:
- There is no longer a one-to-one mapping between `PostgresGoalRecord` and a row in the spec goal table (though, arguably, there never was such a mapping, since the information for `PostgresGoalRecord` is the result of a join between two tables).

I'd be curious to hear your thoughts - the reason I bring this up is I initially picked Option 3 when thinking through this design, but I see you've picked Option 1, and now I'm leaning towards Option 2 as a middle ground 😅.

Reply:
I have made a change which operates as Option 2. I have also taken a note for us to discuss and triage any possible refactoring (as discussed below with @mattdailis and @Twisol ).

Reply:
> * I question, should `enabled` really be the concern of the scheduler?

yes

> Is there anything the scheduler would ever do with a disabled goal, besides ignore it?

It could tell you which goals it skipped because you asked it not to run them. More specifically, it could tell you which goals it could have applied had you not inhibited it.

Reply:
@Twisol do you have any objections to this PR being merged?  I want to make sure I'm giving due time and recognition to the discussions being had here, and that we come to a conclusion, even if that's to create a ticket to take up these improvements.

Reply:
Sorry for the delay (internet trouble), and sorry for the ambiguity. I have no problem with the PR itself; the thing I'm poking at seems to be preexisting.